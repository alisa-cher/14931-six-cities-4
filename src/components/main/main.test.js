import React from 'react';
import MainPage from '../main/main.jsx';
import renderer from 'react-test-renderer';
import {BrowserRouter} from "react-router-dom";

jest.mock(`../map/map.jsx`, () => jest.fn().mockReturnValue(null));

const offers = [{
  type: `appartement`,
  price: 140,
  title: `just amazing1`,
  isPremium: true,
  rating: 4.3,
  previewPhoto: `img/apartment-01.jpg`
},
{
  type: `room`,
  price: 45,
  title: `just amazing2`,
  isPremium: false,
  rating: 4.0,
  previewPhoto: `img/apartment-02.jpg`
}];

const locations = [{
  location: {
    latitude: 1,
    longitude: 2,
    zoom: 0
  },
  name: `Paris`
}];

it(`Main page renders correctly`, () => {
  const tree = renderer
    .create(
        <BrowserRouter>
          <MainPage
            offers={offers}
            locations={locations}
            activeLocation={locations[0]}
            cityCoords={[0, 2]}
            cityZoom={1}
            onCardTitleClick={() => {}}
            onMenuClick={() => {}}/>
        </BrowserRouter>
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});
