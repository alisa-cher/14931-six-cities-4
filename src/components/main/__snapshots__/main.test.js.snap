// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Main page renders correctly 1`] = `
<div
  className="page page--gray page--main"
>
  <header
    className="header"
  >
    <div
      className="container"
    >
      <div
        className="header__wrapper"
      >
        <div
          className="header__left"
        >
          <a
            className="header__logo-link header__logo-link--active"
          >
            <img
              alt="6 cities logo"
              className="header__logo"
              height="41"
              src="img/logo.svg"
              width="81"
            />
          </a>
        </div>
        <nav
          className="header__nav"
        >
          <ul
            className="header__nav-list"
          >
            <li
              className="header__nav-item user"
            >
              <a
                className="header__nav-link header__nav-link--profile"
                href="#"
              >
                <div
                  className="header__avatar-wrapper user__avatar-wrapper"
                >
                  <img
                    alt="user avatar"
                    height="41"
                    style={
                      Object {
                        "borderRadius": "10px",
                      }
                    }
                    width="81"
                  />
                </div>
                <span
                  className="header__user-name user__name"
                />
              </a>
            </li>
          </ul>
        </nav>
      </div>
    </div>
  </header>
  <main
    className="page__main page__main--index"
  >
    <h1
      className="visually-hidden"
    >
      Cities
    </h1>
    <div
      className="tabs"
    >
      <section
        className="locations container"
      >
        <ul
          className="locations__list tabs__list"
        >
          <li
            className="locations__item"
          >
            <a
              className="tabs__item--active locations__item-link tabs__item"
              onClick={[Function]}
            >
              <span>
                Paris
              </span>
            </a>
          </li>
        </ul>
      </section>
    </div>
    <div
      className="cities"
    >
      <div
        className="cities__places-container container"
      >
        <section
          className="cities__places places"
        >
          <h2
            className="visually-hidden"
          >
            Places
          </h2>
          <b
            className="places__found"
          >
            2
             places to stay in 
            Paris
          </b>
          <form
            action="#"
            className="places__sorting"
            method="get"
          >
            <span
              className="places__sorting-caption"
            >
              Sort by
            </span>
            <select
              className="places__sorting-type"
              id="places-sorting"
              onChange={[Function]}
            >
              <option
                className="places__option"
                defaultValue="popular"
                value="popular"
              >
                Popular
              </option>
              <option
                className="places__option"
                defaultValue="popular"
                value="to-high"
              >
                Price: low to high
              </option>
              <option
                className="places__option"
                defaultValue="popular"
                value="to-low"
              >
                Price: high to low
              </option>
              <option
                className="places__option"
                defaultValue="popular"
                value="top-rated"
              >
                Top rated first
              </option>
            </select>
          </form>
          <div
            className="places__list cities__list"
          >
            <article
              className="place-card cities__card"
              onMouseEnter={[Function]}
              onMouseLeave={[Function]}
            >
              <div
                className="place-card__mark"
              >
                <span>
                  Premium
                </span>
              </div>
              <div
                className="cities__image-wrapper place-card__image-wrapper"
              >
                <a
                  href="/property"
                  onClick={[Function]}
                >
                  <img
                    alt="Place image"
                    className="place-card__image"
                    height="200"
                    src="img/apartment-01.jpg"
                    width="260"
                  />
                </a>
              </div>
              <div
                className="place-card__info"
              >
                <div
                  className="place-card__price-wrapper"
                >
                  <div
                    className="place-card__price"
                  >
                    <b
                      className="place-card__price-value"
                    >
                      €
                      140
                    </b>
                    <span
                      className="place-card__price-text"
                    >
                      / night
                    </span>
                  </div>
                  <button
                    className="place-card__bookmark-button button"
                    type="button"
                  >
                    <svg
                      className="place-card__bookmark-icon"
                      height="19"
                      width="18"
                    >
                      <use
                        xlinkHref="#icon-bookmark"
                      />
                    </svg>
                    <span
                      className="visually-hidden"
                    >
                      To bookmarks
                    </span>
                  </button>
                </div>
                <div
                  className="place-card__rating rating"
                >
                  <div
                    className="place-card__stars rating__stars"
                  >
                    <span
                      style={
                        Object {
                          "width": "80%",
                        }
                      }
                    />
                    <span
                      className="visually-hidden"
                    >
                      Rating
                    </span>
                  </div>
                </div>
                <h2
                  className="place-card__name"
                  onClick={[Function]}
                >
                  <a
                    href="/property"
                    onClick={[Function]}
                  >
                    just amazing1
                  </a>
                </h2>
                <p
                  className="place-card__type"
                >
                  appartement
                </p>
              </div>
            </article>
            <article
              className="place-card cities__card"
              onMouseEnter={[Function]}
              onMouseLeave={[Function]}
            >
              <div
                className="cities__image-wrapper place-card__image-wrapper"
              >
                <a
                  href="/property"
                  onClick={[Function]}
                >
                  <img
                    alt="Place image"
                    className="place-card__image"
                    height="200"
                    src="img/apartment-02.jpg"
                    width="260"
                  />
                </a>
              </div>
              <div
                className="place-card__info"
              >
                <div
                  className="place-card__price-wrapper"
                >
                  <div
                    className="place-card__price"
                  >
                    <b
                      className="place-card__price-value"
                    >
                      €
                      45
                    </b>
                    <span
                      className="place-card__price-text"
                    >
                      / night
                    </span>
                  </div>
                  <button
                    className="place-card__bookmark-button button"
                    type="button"
                  >
                    <svg
                      className="place-card__bookmark-icon"
                      height="19"
                      width="18"
                    >
                      <use
                        xlinkHref="#icon-bookmark"
                      />
                    </svg>
                    <span
                      className="visually-hidden"
                    >
                      To bookmarks
                    </span>
                  </button>
                </div>
                <div
                  className="place-card__rating rating"
                >
                  <div
                    className="place-card__stars rating__stars"
                  >
                    <span
                      style={
                        Object {
                          "width": "80%",
                        }
                      }
                    />
                    <span
                      className="visually-hidden"
                    >
                      Rating
                    </span>
                  </div>
                </div>
                <h2
                  className="place-card__name"
                  onClick={[Function]}
                >
                  <a
                    href="/property"
                    onClick={[Function]}
                  >
                    just amazing2
                  </a>
                </h2>
                <p
                  className="place-card__type"
                >
                  room
                </p>
              </div>
            </article>
          </div>
        </section>
        <div
          className="cities__right-section"
        >
          <section
            className="cities__map map"
          />
        </div>
      </div>
    </div>
  </main>
</div>
`;
